core-api/src/main/java/com/optimizely/ab/notification/NotificationCenter.java

```diff
@@ -258,35 +190,27 @@ public void clearAllNotificationListeners() {
      *
      * @param notificationType type of notificationsListeners to remove.
      */
+    @Deprecated
     public void clearNotificationListeners(NotificationType notificationType) {
-        notificationsListeners.get(notificationType).clear();
-    }
-
-    /**
-     * fire a notificaiton of Decision Notification type.
-     *
-     * @param decision containing Decision Notification object
-     */
-    public void sendNotifications(DecisionNotification decision) {
-        for (NotificationHolder holder : decisionListenerHolder) {
-            try {
-                holder.decisionNotificationListener.onDecision(decision);
-            } catch (Exception e) {
-                logger.error("Unexpected exception calling notification listener {}", holder.notificationId, e);
-            }
+        switch (notificationType) {
+            case Track:
+                getNotificationManager(TRACK).clear();
+                break;
+            case Activate:
+                getNotificationManager(ACTIVATE).clear();
+                break;
+            default:
+                throw new OptimizelyRuntimeException("Unsupported notificationType");
         }
     }
 
-    // fire a notificaiton of a certain type.  The arg list changes depending on the type of notification sent.
-    public void sendNotifications(NotificationType notificationType, Object... args) {
-        ArrayList<NotificationHolder> holders = notificationsListeners.get(notificationType);
-        for (NotificationHolder holder : holders) {
-            try {
-                holder.notificationListener.notify(args);
-            } catch (Exception e) {
-                logger.error("Unexpected exception calling notification listener {}", holder.notificationId, e);
-            }
+    @SuppressWarnings("unchecked")
+    public void send(Notification notification) {
+        NotificationManager handler = notifierMap.get(notification.getClass());
+        if (handler == null) {
+            return;
```

Comment:
This seems worth logging a warn or error

Reply:
Replaced with an `OptimizelyRuntimeException` since "we" control the notifications and should be catching this in our CI